client/src/components/Header/components/language-list.tsx

```diff
@@ -0,0 +1,203 @@
+    event: React.MouseEvent<HTMLButtonElement>
+  ): void => {
+    const selectedLanguage = event.currentTarget.dataset.value;
+    const isSelecetedCurrentLanguage =
+      selectedLanguage === clientLocale || selectedLanguage === undefined;
+
+    event.preventDefault();
+    listButtonRef.current?.focus();
+    setShowList(false);
+    if (isSelecetedCurrentLanguage) {
+      return;
+    }
+    const path = createLanguageRedirect({
+      clientLocale,
+      lang: selectedLanguage
+    });
+    if (navigate) {
+      return navigate(path);
+    }
+  };
+
+  const handleMenuKeyDown = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    if (event.key === 'Escape') {
+      listButtonRef.current?.focus();
+      setShowList(false);
+      event.preventDefault();
+    }
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLButtonElement>) => {
+    if (
+      event.relatedTarget &&
+      !event.relatedTarget.closest('.nav-list') &&
+      event.relatedTarget !== listButtonRef.current
+    ) {
+      setShowList(false);
+    }
+  };
+
+  const handleLastLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (!camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const handleFirstLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const getHandleLanguageKeys = (languagePosition: number) => {
+    const lastLanguage = locales.length - 1;
+
+    if (languagePosition === 0) {
+      return handleFirstLangaugeKeys;
+    } else if (languagePosition === lastLanguage) {
+      return handleLastLangaugeKeys;
+    } else handleMenuKeyDown;
+  };
+
+  return pending ? (
+    <div className='nav-skeleton' />
+  ) : (
+    <>
+      <button
+        aria-controls='nav-lang-list'
+        aria-expanded='true'
```

Comment:
`aria-expanded` needs to toggle between `true` and `false` depending on whether the list is showing or not.